src/swarm-lang/Swarm/Language/Typecheck.hs

```diff
@@ -892,11 +895,11 @@ infer s@(CSyntax l t cs) = addLocToTypeErr l $ case t of
   -- each time.
   SApp f x -> do
     -- Infer the type of the left-hand side and make sure it has a function type.
-    f' <- infer f
+    f' <- withFrame l (TCAppL x) $ infer f
     (argTy, resTy) <- decomposeFunTy f (Actual, f' ^. sType)
 
     -- Then check that the argument has the right type.
-    x' <- check x argTy
+    x' <- withFrame l (TCAppR f) $ check x argTy
```

Comment:
I like the comment here better then the LHS/RHS message, as "function argument" doesn't require remembering which side is which. 😅